src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -32,4 +34,22 @@
  */
 public interface Extent extends EntityUniverse, WeatherUniverse, BlockVolume, BiomeArea {
 
+    /**
+     * Get a representation of the block at the given position.
+     *
+     * @param position The position
+     * @return The block
+     */
+    BlockLoc getFullBlock(Vector3i position);
```

Comment:
getBlockExtended is still vague, it should be something more like getBlockLocalized, but I find getFullBlock to be more concise, and since BlockLoc is just the equivalent of the Bukkit Block, devs should understand the idea just fine. Also BlockState isn't a snapshot, that's BlockSnapshot.